fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -305,6 +316,12 @@ def __init__(
         init_start = time.time()
         super().__init__()
         self.process_group = process_group or get_process_group_cached()
+        self.enable_reduce_scatter_overlap = enable_reduce_scatter_overlap
+        if enable_reduce_scatter_overlap:
+            self.process_group_reduce_scatter = process_group_reduce_scatter or get_process_group_cached(
+                ProcessGroupName.reduce_scatter
+            )
+        self.enable_reduce_scatter_overlap = enable_reduce_scatter_overlap
```

Comment:
this is a duplicated line with line 319?

I like how the function `get_process_group_cached` is modified with support multiple names. Thanks for doing that!

To reduce the number of parameters, perhaps the new option can take three values:

None: means using the default group
a PG: means using the giving PG
ProcessGroupName.all_gather: means using the same PG as all_gather

This way, we don't need the enabled option, right?

Reply:
do you mean that the feature is disabled if the value is PG.all_gather?